core/jobs/batch_jobs/audit_rule_validation_jobs.py

```diff
@@ -0,0 +1,2194 @@
+            track of rules denominators uptil now.
+        """
+        invalid_rules = []
+        lower_infinity = float('-inf')
+        upper_infinity = float('inf')
+        for rule_spec in answer_group.rule_specs:
+            if rule_spec.to_dict() in rule_spec_till_now:
+                invalid_rules.append(rule_spec)
+
+            rule_spec_index = str(answer_group.rule_specs.index(
+                rule_spec))
+            range_var: RangeVariable = {
+                'ans_group_index': int(ans_group_index),
+                'rule_spec_index': int(rule_spec_index),
+                'lower_bound': None,
+                'upper_bound': None,
+                'lb_inclusive': False,
+                'ub_inclusive': False
+            }
+            matched_denominator: MatchedDenominator = {
+                'ans_group_index': int(ans_group_index),
+                'rule_spec_index': int(rule_spec_index),
+                'denominator': 0
+            }
+
+            if rule_spec.rule_type in (
+                'IsEquivalentTo', 'IsExactlyEqualTo',
+                'IsEquivalentToAndInSimplestForm'
+            ):
+                rule_value_f = ExpAuditRuleChecksJob._get_rule_value_f(
+                    rule_spec)
+
+                ExpAuditRuleChecksJob._set_lower_and_upper_bounds(
+                        range_var, rule_value_f,
+                        rule_value_f, True, True
+                )
+
+            if rule_spec.rule_type == 'IsGreaterThan':
+                rule_value_f = ExpAuditRuleChecksJob._get_rule_value_f(
+                    rule_spec)
+
+                ExpAuditRuleChecksJob._set_lower_and_upper_bounds(
+                        range_var, rule_value_f,
+                        upper_infinity, False, False
+                )
+
+            if rule_spec.rule_type == 'IsLessThan':
+                rule_value_f = ExpAuditRuleChecksJob._get_rule_value_f(
+                    rule_spec)
+
+                ExpAuditRuleChecksJob._set_lower_and_upper_bounds(
+                        range_var, lower_infinity,
+                        rule_value_f, False, False
+                )
+
+            if rule_spec.rule_type == 'HasDenominatorEqualTo':
+                try:
+                    rule_value_x = int(rule_spec.inputs['x'])
+                    matched_denominator['denominator'] = rule_value_x
+                except Exception:
+                    invalid_rules.append(rule_spec)
+                    continue
+
+            for range_ele in ranges:
+                if ExpAuditRuleChecksJob._is_enclosed_by(
+                    range_var, range_ele
+                ):
+                    earlier_rule = (
+                        answer_groups[range_ele['ans_group_index']]
+                        .rule_specs[range_ele['rule_spec_index']]
+                    )
+                    if ExpAuditRuleChecksJob._should_check_range_criteria(
+                        earlier_rule, rule_spec
+                    ):
```

Comment:
Flip the order of these 2 `if` statements. You should check `is_enclosed_by` only when you should check range criteria, right?

Reply:
Yep!
Done